src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1779,18 +1804,86 @@ RelationRestrictionPartitionKeyIndex(RelationRestriction *relationRestriction)
 
 		if (!targetEntry->resjunk &&
 			IsA(targetExpression, Var) &&
-			IsPartitionColumn(targetExpression, relationPlannerParseQuery))
+			IsPartitionColumn(targetExpression, originalQueryContainingRTEIdentity))
 		{
 			Var *targetColumn = (Var *) targetExpression;
```

Comment:
do we really need `targetColumn` variable? Lets remove either `column` or `targetColumn`, It seems to complicate things.

Reply:
We don't need `column`. removed it.